compiler/scopelang/scope_to_dcalc.ml

```diff
@@ -276,11 +276,28 @@ let rec translate_expr (ctx : ctx) (e : Ast.expr Pos.marked) :
         | _ -> e1_func
       in
       let new_args = List.map (translate_expr ctx) args in
+      let input_typ, output_typ =
```

Comment:
Why did you need the input and output typ here ? Your solution will work for now but it won't be robust when we add functions defined outside scope variables (for instance local let bindings without type information #203 or even anonymous functions). Plus #272 should be merged soon, giving you a way to retrieve the type information in `dcalc` for any sub-expression.

Reply:
Because without them the input and output variables of functions are unembeddable.

For example, with this workaround the generated OCaml code looks like:
```ocaml
(log_end_call
  ["AllocationsFamiliales"; "droit_ouvert_majoration"; "output"]
  (embed_bool) ((log_begin_call ["AllocationsFamiliales"; "droit_ouvert_majoration"]
    droit_ouvert_majoration_)
  ((log_variable_definition ["AllocationsFamiliales"; "droit_ouvert_majoration"; "input"]
    (embed_enfant) param_))))))
```

instead of:
```ocaml
(log_end_call
  ["AllocationsFamiliales"; "droit_ouvert_majoration"; "output"]
  (unembeddable) ((log_begin_call ["AllocationsFamiliales"; "droit_ouvert_majoration"]
    droit_ouvert_majoration_)
  ((log_variable_definition ["AllocationsFamiliales"; "droit_ouvert_majoration"; "input"]
    (unembeddable) param_))))))
```

Reply:
Ok, and maybe refering #272 to be refactored when it's merged?